src/argilla_server/apis/v1/handlers/workspaces.py

```diff
@@ -47,6 +48,21 @@ async def get_workspace(
     return workspace
 
 
+@router.post("/workspaces", status_code=status.HTTP_201_CREATED, response_model=Workspace)
+async def create_workspace(
+    *,
+    db: AsyncSession = Depends(get_async_db),
+    workspace_create: WorkspaceCreate,
+    current_user: User = Security(auth.get_current_user),
+):
+    await authorize(current_user, WorkspacePolicyV1.create)
+
+    if await accounts.get_workspace_by_name(db, workspace_create.name):
+        raise EntityAlreadyExistsError(name=workspace_create.name, type=Workspace)
```

Comment:
@frascuchon I will move this block of code to the context or the validator if we think it's ok.

Reply:
Done.